lib/services/authority.go

```diff
@@ -86,6 +88,34 @@ func checkUserOrHostCA(cai types.CertAuthority) error {
 	return trace.Wrap(err)
 }
 
+// checkDatabaseCA checks if provided certificate authority contains a valid TLS key pair.
+// This function is used to verify Database CA.
+func checkDatabaseCA(cai types.CertAuthority) error {
+	ca, ok := cai.(*types.CertAuthorityV2)
+	if !ok {
+		return trace.BadParameter("unknown CA type %T", cai)
+	}
+
+	if len(ca.Spec.ActiveKeys.TLS) == 0 && len(ca.Spec.TLSKeyPairs) == 0 {
+		return trace.BadParameter("DB certificate authority missing TLS key pairs")
+	}
+
+	for _, pair := range ca.GetTrustedTLSKeyPairs() {
+		if len(pair.Key) > 0 && pair.KeyType == types.PrivateKeyType_RAW {
+			_, err := utils.ParsePrivateKey(pair.Key)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		}
+		_, err := tlsca.ParseCertificatePEM(pair.Cert)
```

Comment:
I would probably consider using x509.LoadTLSKeyPair for make sure that cert/key form a valid key pair instead of just parsing them individually.

Reply:
Added above.